forest/jasmine/traj2stats.py

```diff
@@ -819,121 +862,242 @@ def gps_quality_check(study_folder: str, study_id: str) -> float:
+        and logger csv file to show warnings and bugs during the run
     """
-    
-    quality_threshold = quality_threshold if quality_threshold is not None else 0.05
-    
-    if os.path.exists(output_folder)==False:
-        os.mkdir(output_folder)
-
-    if parameters == None:
-        parameters = [60*60*24*10,60*60*24*30,0.002,200,5,1,0.3,0.2,0.5,100,0.01,0.05,3,10,2,'GLC',10,51,None,None,None]
-    [l1,l2,l3,g,a1,a2,b1,b2,b3,d,sigma2,tol,switch,num,linearity,method,itrvl,accuracylim,r,w,h] = parameters
-    pars0 = [l1,l2,l3,a1,a2,b1,b2,b3]
-    pars1 = [l1,l2,a1,a2,b1,b2,b3,g]
-
-    if r == None:
-        orig_r = None
-    if w == None:
-        orig_w = None
-    if h == None:
-        orig_h = None
-
-    ## beiwe_id should be a list of str
-    if beiwe_id == None:
-        beiwe_id = os.listdir(study_folder)
-    ## create a record of processed user ID and starting/ending time
-
-    if all_memory_dict == None:
+
+    os.makedirs(output_folder, exist_ok=True)
+
+    if parameters is None:
+        parameters = Hyperparameters()
+
+    pars0 = [
+        parameters.l1, parameters.l2, parameters.l3, parameters.a1,
+        parameters.a2, parameters.b1, parameters.b2, parameters.b3
+    ]
+    pars1 = [
+        parameters.l1, parameters.l2, parameters.a1, parameters.a2,
+        parameters.b1, parameters.b2, parameters.b3, parameters.g
+    ]
+
+    orig_r = parameters.r
+    orig_w = parameters.w
+    orig_h = parameters.h
+
+    # beiwe_study_ids should be a list of str
+    if beiwe_study_ids is None:
+        beiwe_study_ids = os.listdir(study_folder)
+    # create a record of processed user study_id and starting/ending time
+
+    if all_memory_dict is None:
         all_memory_dict = {}
-        for ID in beiwe_id:
-            all_memory_dict[str(ID)] = None
-
-    if all_BV_set == None:
-        all_BV_set = {}
-        for ID in beiwe_id:
-            all_BV_set[str(ID)] = None
-
-    if option == 'both':
-        if os.path.exists(output_folder+"/hourly")==False:
-            os.mkdir(output_folder+"/hourly")
-        if os.path.exists(output_folder+"/daily")==False:
-            os.mkdir(output_folder+"/daily")
-
-    if save_traj == True:
-        if os.path.exists(output_folder+"/trajectory")==False:
-            os.mkdir(output_folder+"/trajectory")
-
-    if len(beiwe_id)>0:
-        for ID in beiwe_id:
-            sys.stdout.write('User: '+ ID  + '\n')
+        for study_id in beiwe_study_ids:
+            all_memory_dict[str(study_id)] = None
+
+    if all_bv_set is None:
+        all_bv_set = {}
+        for study_id in beiwe_study_ids:
+            all_bv_set[str(study_id)] = None
+
+    if frequency == Frequency.BOTH:
+        os.makedirs(f"{output_folder}/hourly", exist_ok=True)
+        os.makedirs(f"{output_folder}/daily", exist_ok=True)
+    if save_traj:
+        os.makedirs(f"{output_folder}/trajectory", exist_ok=True)
+
+    if len(beiwe_study_ids) > 0:
```

Comment:
I think this line is redundant so can be removed. This will also free up some line space in the code below when the for loop is shifted left.